File: 32/cpp/main.cc
#include <cctype>
#include <iostream>

#include "guess_game.h"
#include "user_input.h"

namespace {

bool user_want_to_continue() {
    auto answer = read_user_input("Play again? ");
    return toupper(answer[0]) == 'Y';
}

}  // namespace

int main() {
    for (;;) {
        guessgame::start_game();

        if (!user_want_to_continue()) {
            break;
        }
    }
    std::cout << "Goodbye!" << std::endl;

    return 0;
}

File: 32/go/main.go
package main

import (
	"fmt"
	"strings"
)

func userWantToContinue() bool {
	answer := readUserInput("Play again? ")
	return strings.ToUpper(answer)[0] == 'Y'
}

func main() {
	for {
		startGame()
		if !userWantToContinue() {
			break
		}
	}
	fmt.Println("Goodbye!")
}
